src/hera/workflows/_meta_mixins.py

```diff
@@ -643,21 +658,25 @@ def script_decorator(func: Callable[FuncIns, FuncR]) -> Callable:
             def script_call_wrapper(*args, **kwargs) -> Union[FuncR, Step, Task, None]:
                 """Invokes a CallableTemplateMixin's `__call__` method using the given SubNode (Step or Task) args/kwargs."""
                 if _context.declaring:
-                    try:
-                        # ignore decorator function assignment
-                        subnode_name = varname()
-                    except ImproperUseError:
-                        # Template is being used without variable assignment (so use function name or provided name)
-                        subnode_name = script_template.name  # type: ignore
-
-                    subnode_name = kwargs.pop("name", subnode_name)
-                    assert isinstance(subnode_name, str)
+                    if "name" in kwargs:
+                        subnode_name = kwargs.pop("name")
+                    else:
```

Comment:
`subnode_name` will still have an `Any` type from the `kwargs.pop` - I'm not sure it's worth adding explicit type checking/handling around this (this is Python after all). Will try some variations out but will likely leave the code as-is.

Reply:
The issue isn't the `Any` type, it's that if the user does an explicit `name=None`, which our type annotations say we allow, and which logically should mean the same thing as leaving it off, we erroneously use the string `"None"` as the name.

Reply:
This is admittedly niche. I'm a fan of fixing it anyway because it simplifies the code. It's most likely to come up if someone has parameterized code, e.g.:

```python
def add_script(workflow, name = None):
  @workflow.script(name=name)
  def my_step():
      ...
```